File: superstitioMain/java/superstitio/cards/maso/PowerCard/BodyModification_SuperHugeBreast.java
package superstitio.cards.maso.PowerCard;

import basemod.helpers.CardModifierManager;
import com.megacrit.cardcrawl.cards.DamageInfo;
import com.megacrit.cardcrawl.characters.AbstractPlayer;
import com.megacrit.cardcrawl.monsters.AbstractMonster;
import superstitio.DataManager;
import superstitio.cardModifier.modifiers.card.BodyModificationTag;
import superstitio.cards.SuperstitioCard;
import superstitio.cards.maso.MasoCard;
import superstitio.powers.EasyBuildAbstractPowerForPowerCard;
import superstitio.powers.Milk;

public class BodyModification_SuperHugeBreast extends MasoCard {
    public static final String ID = DataManager.MakeTextID(BodyModification_SuperHugeBreast.class);

    public static final CardType CARD_TYPE = CardType.POWER;

    public static final CardRarity CARD_RARITY = CardRarity.UNCOMMON;

    public static final CardTarget CARD_TARGET = CardTarget.SELF;

    private static final int COST = 2;
    private static final int MAGIC = 4;
    private static final int UPGRADE_MAGIC = 2;

    public BodyModification_SuperHugeBreast() {
        super(ID, CARD_TYPE, COST, CARD_RARITY, CARD_TARGET);
        this.setupMagicNumber(MAGIC, UPGRADE_MAGIC);
        CardModifierManager.addModifier(this, new BodyModificationTag());
    }

    @Override
    public void use(AbstractPlayer player, AbstractMonster monster) {
        addToBot_applyPower(new BodyModification_SuperHugeBreastPower(this.magicNumber));
    }

    @Override
    public void upgradeAuto() {
    }

    public static class BodyModification_SuperHugeBreastPower extends EasyBuildAbstractPowerForPowerCard {

        public BodyModification_SuperHugeBreastPower(int amount) {
            super(amount);
        }

        @Override
        public int onAttacked(DamageInfo info, int damageAmount) {
            if (info.type != DamageInfo.DamageType.NORMAL) return damageAmount;
            addToBot_applyPower(new Milk(owner, this.amount));
            return damageAmount;
        }

        @Override
        public void updateDescriptionArgs() {
            setDescriptionArgs(this.amount);
        }

        @Override
        protected SuperstitioCard makePowerCard() {
            return new BodyModification_SuperHugeBreast();
        }
    }
}

File: superstitioMain/java/superstitio/cards/maso/PowerCard/BodyModification_SuperHugeBreast.kt
package superstitio.cards.maso.PowerCard

import basemod.helpers.CardModifierManager
import com.megacrit.cardcrawl.cards.DamageInfo
import com.megacrit.cardcrawl.cards.DamageInfo.DamageType
import com.megacrit.cardcrawl.characters.AbstractPlayer
import com.megacrit.cardcrawl.monsters.AbstractMonster
import superstitio.DataManager
import superstitio.cardModifier.modifiers.card.BodyModificationTag
import superstitio.cards.SuperstitioCard
import superstitio.cards.maso.MasoCard
import superstitio.powers.EasyBuildAbstractPowerForPowerCard
import superstitio.powers.Milk
import superstitioapi.utils.addToBot_applyPower
import superstitioapi.utils.setDescriptionArgs

class BodyModification_SuperHugeBreast : MasoCard(ID, CARD_TYPE, COST, CARD_RARITY, CARD_TARGET)
{
    init
    {
        this.setupMagicNumber(MAGIC, UPGRADE_MAGIC)
        CardModifierManager.addModifier(this, BodyModificationTag())
    }

    override fun use(player: AbstractPlayer?, monster: AbstractMonster?)
    {
        addToBot_applyPower(BodyModification_SuperHugeBreastPower(this.magicNumber))
    }

    override fun upgradeAuto()
    {
    }

    class BodyModification_SuperHugeBreastPower(amount: Int) : EasyBuildAbstractPowerForPowerCard(amount)
    {
        override fun wasHPLost(info: DamageInfo, damageAmount: Int)
        {
            if (info.type != DamageType.NORMAL) return
            addToBot_applyPower(Milk(owner, this.amount))
        }

        override fun updateDescriptionArgs()
        {
            setDescriptionArgs(this.amount)
        }

        override fun makePowerCard(): SuperstitioCard
        {
            return BodyModification_SuperHugeBreast()
        }
    }

    companion object
    {
        val ID: String = DataManager.MakeTextID(BodyModification_SuperHugeBreast::class.java)

        val CARD_TYPE: CardType = CardType.POWER

        val CARD_RARITY: CardRarity = CardRarity.UNCOMMON

        val CARD_TARGET: CardTarget = CardTarget.SELF

        private const val COST = 2
        private const val MAGIC = 4
        private const val UPGRADE_MAGIC = 2
    }
}
